dashboard/app/models/census/census_summary.rb

```diff
@@ -28,4 +28,161 @@ class Census::CensusSummary < ApplicationRecord
   enum teaches_cs: TEACHES
 
   validates_presence_of :audit_data
+
+  def self.submission_teaches_cs(submission, is_high_school)
+    if is_high_school
+      (submission.how_many_20_hours_some? || submission.how_many_20_hours_all?)
+    else
+      (
+        submission.how_many_10_hours_some? ||
+        submission.how_many_10_hours_all? ||
+        submission.how_many_20_hours_some? ||
+        submission.how_many_20_hours_all?
+      )
+    end
+  end
+
+  def self.summarize_census_data
+    latest_survey_year = Census::CensusSubmission.maximum(:school_year)
+    latest_ap_data_year = Census::ApCsOffering.maximum(:school_year)
+    latest_ib_data_year = Census::IbCsOffering.maximum(:school_year)
+    latest_state_data_years = {}
+    state_years_with_data = {}
+    Census::StateCsOffering::SUPPORTED_STATES.each do |state|
+      state_years_with_data[state] = Census::StateCsOffering.
+                                       joins(:school).
+                                       where(schools: {state: state}).
+                                       select(:school_year).
+                                       group(:school_year).
+                                       map(&:school_year)
+      latest_state_data_years[state] = state_years_with_data[state].max
+    end
+
+    max_year = [
+      latest_survey_year,
+      latest_ap_data_year,
+      latest_ib_data_year,
+      latest_state_data_years.map {|_, v| v}.max
```

Comment:
What does this do? AFAIK `map` for an array will only provide a single parameter to the block. Are you missing a key, or is this meant to be something else?

```ruby
irb(main):001:0> [1,2].map{|_,v| v}
=> [nil, nil]
```

Reply:
`latest_state_data_years` is a hash, not any array. The keys are the state code and the value is the latest year that has data for that state. This is taking the latest year across all states that have data. This would be clearer if I rename `latest_state_data_years` to ` latest_data_year_by_state` and then do `latest_data_year_by_state.values.max`

Reply:
OIC. That makes sense. I misread that anyway and thought the map was outside the array. Maybe add a small comment (or rename as you suggested).

Reply:
I cleaned this up a bit. Is it clearer now?

Reply:
Much better. Thanks!